modules/matchRoutes.js

```diff
@@ -44,13 +44,13 @@ function getIndexRoute(route, location, callback) {
 }
 
 function assignParams(params, paramNames, paramValues) {
-  return paramNames.reduceRight(function (params, paramName, index) {
+  return paramNames.reduce(function (params, paramName, index) {
```

Comment:
`reduceRight` uses the last params first, so subsequent params that are named the same thing appear after previous values in the params array.